test/one/profiler/test/SkipReason.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package one.profiler.test;
+
+public enum SkipReason {
+    Disabled,
+    ArchMismatch,
+    OsMismatch,
+    JvmMismatch,
+    JvmVersionMismatch,
+    SkipByClassName,
+    SkipByMethodName,
+}
```

Comment:
Two-level test status is bit of overengineering, IMO. Much code for a small (although useful) feature.
I don't think there is much value in differentiating tests skipped by a class name or a method name (especially since we'll likely unify this with a regex or a glob pattern). Difference between OS/Arch and Jvm/JvmVersion is also subtle. The only difference that probably makes sense is between not applicable tests vs. manually disabled tests. We can have two TestStatuses for them, or we can simply exclude not applicable tests from the total count during the first pass.
Actually, the previous version was already good. I'd just address the problem of 100% pass rate with one of the approaches mentioned above. Wdyt?

Reply:
I made the skip details visible, because in every configuration there would be some skipped tests. While in the latest code this doesn't affect the 100% pass rate calculation, it may raise suspicion to someone reading the run results.

I like the idea of having two types of skip, manual and config mismatch.

Reply:
Summary would look like:

```
Results Summary:
PASS: 35
SKIP (disabled): 0
SKIP (config mismatch): 3
FAIL: 0
TOTAL: 38 (100.00% PASS)
```

Reply:
Yes! Just a minor reordering, maybe?
```
Results Summary:
PASS: 35 (100%)
FAIL: 0
SKIP (disabled): 0
SKIP (config mismatch): 3
TOTAL: 38
```